src/ServiceStack.Mvc/ServiceStackController.cs

```diff
@@ -53,7 +53,17 @@ public virtual ActionResult AuthenticationErrorResult
             get
             {
                 var returnUrl = HttpContext.Request.GetPathAndQuery();
-                return new RedirectResult(UnauthorizedRedirectUrl.Fmt(returnUrl.UrlEncode()));
+
+                var encoded = returnUrl.UrlEncode();
+                var unauthorizedRedirectUrl = UnauthorizedRedirectUrl;
+
+                if (string.IsNullOrEmpty(unauthorizedRedirectUrl))
+                {
+                    throw new HttpException(401, "Unauthorized");
+                }
+
+                var formatted = unauthorizedRedirectUrl.Fmt(encoded);
```

Comment:
You removed `?redirect={0}#f=Unauthorized`, where you using string.format encoded after.. Even though no error will be throwing, but what's the point? And by this change you removed default behavior. 

Things can be improve is under AuthFeature class's GetHtmlRedirect.
```
if (feature != null)
                return feature.HtmlRedirect;
```
to 
```
if (feature != null && feature.HtmlRedirect != null)
                return feature.HtmlRedirect;
```
Also HtmlRedirect did setup automatically, unless you change that value after AuthFeature been created.